### ISSUE TITLE
Issue with Affiliate Links data

### ISSUE THREAD

**Issue by HardeepAsrani** [Codeinwp/wp-product-review#289]
If you give both button the same text, only one will appear in the front-end. It happens because it uses the name value as the key, instead having `name` as the key and name's value as the value of the key.

It's making it really hard for me to use this data in Gutenberg, as stdClasses aren't JSON friendly, and having the value as a key makes it even harder.

Can we get it fixed soon so I could implement it in the Gutenberg block properly? @selul @rodica-andronache

**Comment by HardeepAsrani** [comment]
If the description sounds confusing, because it is, then some visual aid of how it is, and how it should be. :D

![screenshot-wppr local-2018-05-25-16-16-11](https://user-images.githubusercontent.com/2649903/40540839-2e9618ce-6037-11e8-974e-6ffb55e9ed0e.png)

**Comment by rodica-andronache** [comment]
@HardeepAsrani I didn't understand :D There are options for two buttons? I only know about one.

**Comment by HardeepAsrani** [comment]
Yup, there’s a plus button that you need to click for the second to appear. Instead of saving the name and link as values of two properties, it’s saving link as the value of name property.

**Comment by rodica-andronache** [comment]
@HardeepAsrani thanks, I understand now :D 
@contactashish13 could you please check this? thanks

**Comment by contactashish13** [comment]
@HardeepAsrani is it even a valid scenario where two buttons will have the same text but might have different links?

**Comment by HardeepAsrani** [comment]
You can have two links from Amazon. Regardless of the scenario, having this structure is making it hard to manage the state in Gutenberg block as its JSON output isn’t much friendly. So if we could have it as a pair of name and value, like wppr_options, it would be great. @contactashish13

**Comment by contactashish13** [comment]
@HardeepAsrani maybe you can use the filter wppr_links to change the format to what you want for gutenberg? Changing this format in the code might jeopardize existing installs.

`apply_filters( 'wppr_links', $this->links, $this->ID );`
